temp.ts

```diff
@@ -0,0 +1,136 @@
+            default: () => true,
+          }),
+        },
+        filter_relationship: {
+          is_active: 'true',
+        },
+      }),
+      name: Field.String({ required: true }),
+      is_active: Field.Boolean({
+       default: () => true,
+     }),
+   }
+   super(values, labels, attributes)
+ }
+ serialize() {
+   const addSuffix = (path: string, suffix: string) => {
+     const splittedPath = path.split('.')
+     return splittedPath.slice(0, -1).join('.') + suffix + '.' + splittedPath[splittedPath.length - 1]
+   }
+   return {
+     id: this.id,
+     email: this.email,
+     name: this.name,
+     profile_image: this.profile_image
+       ? {
+           original: this.profile_image.path,
+           '200w': addSuffix(this.profile_image.path, '-small'),
+           '100w': addSuffix(this.profile_image.path, '-smallest'),
+         }
+       : undefined,
+   }
+ }
+ static async findByIdOrFail(id: number) {
+   const caker = await Caker.findByID(id, { with_related: ['profile_image'] })
+   if (!caker) {
+     throw new ResourceNotFoundException(messages.CAKER_NOT_FOUND)
+   }
+   return caker as Caker
+ }
+ static async findByEmail(email: string): Promise<Caker | null> {
+   const caker = (await Caker.findBy(
+     [
+       {
+         key: 'email',
+         value: email,
+       },
+       {
+         key: 'is_active',
+         value: true,
+       },
+     ],
+     { with_related: ['profile_image'] }
+   )) as Collection<Caker>
+   return caker.first() as Caker | null
+ }
+  static findByFilter(
+    params: Partial<Pick<CakerType, 'email' | 'id' | 'is_active' | 'name'>>
+  ): Promise<Collection<Caker>> {
+    const filter = Object.entries(params).reduce((acc: FilterAttributes, [key, value]) => {
+      if (value === undefined) {
+        return acc
+      }
+      if (key === 'id') {
+        return [...acc, { key: 'id(caker)', value }]
+      }
+
+      return [...acc, { key, value, operator: typeof value === 'string' ? 'STARTS WITH' : '=' }]
+    }, [])
+    return Caker.findBy(filter, { with_related: ['profile_image'] }) as Promise<Collection<Caker>>
+  }
+
+  static async softDeleteById(id: number) {
+    const caker = await Caker.findByIdOrFail(id)
+    caker.is_active = false
+    await caker.save()
+    return caker
+  }
+
+  static async updateProfileImage(caker: Caker, newProfileImage: Express.Multer.File) {
+    const file = await File.createFromFile(newProfileImage)
+    if (caker.profile_image) {
+      await caker.updateRelationship('profile_image', caker.profile_image, { is_active: false })
+    }
+    await caker.createRelationship('profile_image', file)
+    return caker
+  }
+
+  static async create({ email, name, profileImage }: CreateArgs) {
+    const caker = new Caker({
+      email,
+      name,
+    })
+    await caker.save()
+    const file = await File.createFromFile(profileImage)
+    await caker.createRelationship('profile_image', file)
+    return caker
+  }
+}
+
+export { Caker }
```

Comment:
ChatGPT Review:

    The code above is for a model class for a caker. A caker is a user of the app who can create and rate cakes.
    The model class defines the properties of a caker, as well as some methods for interacting with cakers.
    
    There are no major bugs in the code. However, there are a few minor issues:
    
    - The indentation is not consistent throughout the code. This can make the code difficult to read.
    - There is no documentation for the code. Documentation would be helpful for other developers who need to work with this code.
    - The naming of some variables could be improved. For example, the variable 'file' is used to store a caker's profile image. A more descriptive name would be helpful.